internal/generator/generator.go

```diff
@@ -75,13 +75,15 @@ func constructRepositorySpec(pkg *types.Package, namedStruct *types.Named,
 }
 
 func constructCodeBuilder(pkg *types.Package, namedStruct *types.Named,
-	interfaceName string, methodSpecs []spec.MethodSpec) (*codegen.Builder, error) {
+	interfaceName string, methodSpecs []spec.MethodSpec, outputPkgName string) (*codegen.Builder, error) {
 
 	generator := mongo.NewGenerator(pkg, namedStruct, interfaceName)
```

Comment:
`pkg` here should be the target output package. It will be used to determine whether to write named type as `User` or `user.User`

Reply:
sorry I was missing that part. 

Do you think we need to create  `outputPkg := types.NewPackage(outputPkgPath, outputPkgName)` ?

in that case it might need path and name...

Reply:
Is it possible to receive package path in the command line arguments instead of only the package name? Package paths should be able to derive package names.

Reply:
According to your requirement, do you require that the generator automatically creates a new package if not exists? We can do two approaches:

1. If we will not automatically create new packages when no Go files exist, we can change the command line arg dest-pkg to be the package path instead of only the package name.
2. If we need to automatically create new packages when no Go files exist, we might need another command line arg.

But for simplicity, I prefer that we do not automatically create new packages. Are there any concerns?

Reply:
I understand with your concern and I am OK going with option 1. 

3f27161ee34c308c33f0474b593ff1fde024ce8d